### PR TITLE
add mimir support for resource usage estimation recording rules

### PR DESCRIPTION
**Pull request by QuantumEnigmaa**
Towards: https://github.com/giantswarm/giantswarm/issues/30857

This PR adds mimir support for the resource usage estimation recording rules. Instead of measuring the resource usage impact on a prometheus instance, it will now be measured on the mimir-ingester pods.

This PR : https://github.com/giantswarm/shared-configs/pull/86 needs to be merged prior to this one so that the query doesn't fail with mimit.

### Checklist

- [x] Update CHANGELOG.md
- [ ] Add [Unit tests](https://github.com/giantswarm/prometheus-rules/#testing)
- [x] Follow [Alert structure](https://github.com/giantswarm/prometheus-rules/#how-alerts-are-structured)
- [ ] Consider [creating a dashboard](https://docs.giantswarm.io/getting-started/observability/grafana/custom-dashboards/) ([guidelines](https://intranet.giantswarm.io/docs/product/ux/guidelines/dashboards/)) (if it does not exist already) to help oncallers monitor the status of the issue.
- [ ] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).

### CODE REVIEW

File: helm/prometheus-rules/templates/platform/atlas/recording-rules/monitoring.resource-usage-estimation.rules.yaml [hunk 1]
```diff
@@ -1,3 +1,20 @@
...
+metadata:
+  labels:
+    {{- include "labels.common" . | nindent 4 }}
+  name: monitoring.resource-usage-estimation.recording.rules
+  namespace: {{ .Values.namespace  }}
+spec:
+  groups:
+  - name: monitoring.resource-usage-estimation.recording
+    rules:
+    - expr: (count({__name__=~".+"}) by (cluster_id, job) / on(cluster_id) group_left prometheus_tsdb_head_series{instance=~"^.*:9090"}) * on(cluster_id) group_left sum(container_memory_usage_bytes{container="ingester", namespace="mimir"}) by (cluster_id)
```
Comment (QuentinBisson) [inline helm/prometheus-rules/templates/platform/atlas/recording-rules/monitoring.resource-usage-estimation.rules.yaml:13]: You can add the if only here and not duplicate most of the template :D

Comment (QuentinBisson) [reply helm/prometheus-rules/templates/platform/atlas/recording-rules/monitoring.resource-usage-estimation.rules.yaml:13]: Does Mimir actually have prometheus_tsdb_head_series ?

Comment (QuantumEnigmaa) [reply helm/prometheus-rules/templates/platform/atlas/recording-rules/monitoring.resource-usage-estimation.rules.yaml:13]: Yeah you're right !

Comment (QuantumEnigmaa) [reply helm/prometheus-rules/templates/platform/atlas/recording-rules/monitoring.resource-usage-estimation.rules.yaml]: Fixed both.

**Review by QuentinBisson — APPROVED** [review]
LGTM <3

**Comment by QuantumEnigmaa** [comment]
@hervenicol WDYT ? In any case I'll wait for https://github.com/giantswarm/shared-configs/pull/86 to be merged before merging this one.

File: helm/prometheus-rules/templates/platform/atlas/recording-rules/monitoring.resource-usage-estimation.rules.yaml [hunk 2]
```diff
@@ -9,7 +9,14 @@ spec:
   groups:
   - name: monitoring.resource-usage-estimation.recording
     rules:
+    {{- if .Values.mimir.enabled }}
+    - expr: (count({__name__=~".+"}) by (cluster_id, job) / on(cluster_id) group_left cortex_ingester_active_series{container="ingester"}) * on(cluster_id) group_left sum(container_memory_usage_bytes{container="ingester", namespace="mimir"}) by (cluster_id)
```
Comment (hervenicol) [inline helm/prometheus-rules/templates/platform/atlas/recording-rules/monitoring.resource-usage-estimation.rules.yaml:13]: Shouldn't we sum `cortex_ingester_active_series{container="ingester"}`?

**Comment by hervenicol** [comment]
It was quite a heavy query on prometheus, but I don't know how it will behave on mimir where we have metrics for all clusters.
Did you try "unlocking" Grizzly to test it there? And have an idea of how heavy the query is?

File: helm/prometheus-rules/templates/platform/atlas/recording-rules/monitoring.resource-usage-estimation.rules.yaml (hunk 2 above)
Comment (QuantumEnigmaa) [reply helm/prometheus-rules/templates/platform/atlas/recording-rules/monitoring.resource-usage-estimation.rules.yaml]: You're right, in our case we'd better sum as it'll all about the ingesters anyway. 
Updated the query.

**Comment by QuantumEnigmaa** [comment]
> Did you try "unlocking" Grizzly to test it there?

What do you mean ?

> And have an idea of how heavy the query is?

Unfortunately no, I didn't find the query's size mentioned anywhere :/

**Comment by hervenicol** [comment]
> > Did you try "unlocking" Grizzly to test it there?
> 
> What do you mean ?

Updating the limits so that the query could run, as per https://github.com/giantswarm/shared-configs/pull/86

**Comment by hervenicol** [comment]
I think the query could be lighter by using `scrape_samples_post_metric_relabeling` rather than `{__name__}`.

You should look at the `ServiceMonitors Overview` / `Samples scraped per servicemonitor` dashboard for how to use it.

**Comment by QuentinBisson** [comment]
I agree it is wayy faster than the count :D

**Comment by QuantumEnigmaa** [comment]
Tested and working on `grizzly` : 
image

**Comment by QuentinBisson** [comment]
@QuantumEnigmaa is this faster to execute?
 Maybe we should also use this metric for vintage instead of the count as we also have it

**Review by QuentinBisson — APPROVED** [review]

**Comment by QuantumEnigmaa** [comment]
@hervenicol All good for you ?

**Review by hervenicol — APPROVED** [review]
LGTM

**Pull request merged**